cmd/ping.go

```diff
@@ -39,55 +42,98 @@ Examples:
   ping jsdelivr.com from aws+montreal --latency
 
   # Ping jsdelivr.com from a probe in ASN 123 with json output
-  ping jsdelivr.com from 123 --json`,
+  ping jsdelivr.com from 123 --json
+  
+  # Continuously ping google.com from New York
+  ping google.com from New York --infinite`,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		// Create context
 		err := createContext(cmd.CalledAs(), args)
 		if err != nil {
 			return err
 		}
+		if ctx.Infinite {
+			return infinitePing(cmd)
+		}
+		_, err = ping(cmd)
+		return err
+	},
+}
+
+func infinitePing(cmd *cobra.Command) error {
+	var err error
+	if ctx.Limit > 5 {
+		return fmt.Errorf("continous mode is currently limited to 5 probes")
+	}
+	ctx.Packets = 16 // Default to 16 packets
+
+	// Trap sigterm or interupt to display info on exit
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
 
-		// Make post struct
-		opts = model.PostMeasurement{
-			Type:              "ping",
-			Target:            ctx.Target,
-			Limit:             ctx.Limit,
-			InProgressUpdates: inProgressUpdates(ctx.CI),
-			Options: &model.MeasurementOptions{
-				Packets: packets,
-			},
+	go func() {
+		for {
+			ctx.From, err = ping(cmd)
```

Comment:
Seems the history handling broke here - now when `--infinite` makes 3 requests in the background, there are 3 entries in the history (should be just one).